src/main/java/ch/njol/skript/events/EvtPeriodical.java

```diff
@@ -18,132 +18,109 @@
  */
 package ch.njol.skript.events;
 
-import org.bukkit.Bukkit;
-import org.bukkit.World;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-
 import ch.njol.skript.Skript;
 import ch.njol.skript.SkriptEventHandler;
 import ch.njol.skript.events.bukkit.ScheduledEvent;
 import ch.njol.skript.events.bukkit.ScheduledNoWorldEvent;
 import ch.njol.skript.lang.Literal;
-import ch.njol.skript.lang.SelfRegisteringSkriptEvent;
+import ch.njol.skript.lang.SkriptEvent;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.Trigger;
 import ch.njol.skript.util.Timespan;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtPeriodical extends SkriptEvent {
 
-/**
- * @author Peter Güttinger
- */
-public class EvtPeriodical extends SelfRegisteringSkriptEvent {
 	static {
 		Skript.registerEvent("*Periodical", EvtPeriodical.class, ScheduledNoWorldEvent.class, "every %timespan%")
 				.description("An event that is called periodically.")
-				.examples("every 2 seconds:",
-						"every minecraft hour:",
-						"every tick: # can cause lag depending on the code inside the event",
-						"every minecraft days:")
-				.since("1.0");
+				.examples(
+					"every 2 seconds:",
+					"every minecraft hour:",
+					"every tick: # can cause lag depending on the code inside the event",
+					"every minecraft days:"
+				).since("1.0");
 		Skript.registerEvent("*Periodical", EvtPeriodical.class, ScheduledEvent.class, "every %timespan% in [world[s]] %worlds%")
 				.description("An event that is called periodically.")
-				.examples("every 2 seconds in \"world\":",
-						"every minecraft hour in \"flatworld\":",
-						"every tick in \"world\": # can cause lag depending on the code inside the event",
-						"every minecraft days in \"plots\":")
-				.since("1.0")
+				.examples(
+					"every 2 seconds in \"world\":",
+					"every minecraft hour in \"flatworld\":",
+					"every tick in \"world\": # can cause lag depending on the code inside the event",
+					"every minecraft days in \"plots\":"
+				).since("1.0")
 				.documentationID("eventperiodical");
 	}
 	
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Timespan period;
-	
-	@Nullable
-	private Trigger t;
-	@Nullable
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private int[] taskIDs;
-	
-	@Nullable
-	private transient World[] worlds;
-	
-//	@Nullable
-//	private String[] worldNames;
+
+	private World @Nullable [] worlds;
 	
 	@SuppressWarnings("unchecked")
 	@Override
```

Comment:
While you're here \:)
```suggestion
	@Override
	@SuppressWarnings("unchecked")
```